### ISSUE TITLE
Blacklist image hashes?

### ISSUE THREAD

**Issue by frozenpandaman** [vichan-devel/vichan#651]
Sorry for all the questions/issues lately. Is there a recommended way to blacklist certain image hashes? Hoping this will cut down on bot spam that seemingly posts the same things over and over.

I'm guessing there would naīvely be a way just to hardcode it into some .php file (if image hash in_array(whatever1, whatever2), return without doing anything or serve an error message) but I'm not sure where the image upload logic is happening.

**Comment by RealAngeleno** [comment]
Yes. Do something like this:

`$config['filters'][] = array(
    'condition' => array(
               'custom' => function($post) {
                       if ( array_key_exists('filehash',$post) && in_array($post['filehash'],array(
'your filehash',

)))
return true;
else
return false;}),
'action' => 'ban',
'add_note' => true,
'all_boards' => true,
'expires' => 60 * 60 * 72, // Three Days
'reason' => 'Ban evasion.'
);`

Kuz hacked up a script to make it so that mods can see the md5 hashes, but he never made it public. Best way to find it is by checking the filehash column on the posts_[board] table.

**Comment by frozenpandaman** [comment]
@RealAngeleno @crazy4cars69 Thank you both so much! Really appreciated.

I'm not too great with PHP myself so being able to implement this was extremely helpful. Do let me know if you'd happen to be able to throw together some simple QuestyCaptcha code (i.e. just an input 'verification' field that checks the entered string against something in config.php) if you have time in the future.

**Comment by frozenpandaman** [comment]
Unfortunately this doesn't seem to be helping with spam issues. Spambots have the same image saved presumably at different compression levels or with different metadata or something, causing the blacklist for specific hashes to be ineffective.

**Comment by RealAngeleno** [comment]
I was asked a bit about questycaptcha from a few other people too. That'll be my top priority for now, along with the wiki.

**Comment by RealAngeleno** [comment]
Though I wouldn't know the best way to implement it, as there's many ways to do it.

**Comment by frozenpandaman** [comment]
> I was asked a bit about questycaptcha from a few other people too. That'll be my top priority for now

@RealAngeleno AMAZING to hear, thank you!!!

**Comment by frozenpandaman** [comment]
> I was asked a bit about questycaptcha from a few other people too. That'll be my top priority for now, along with the wiki.

Just wanted to ask if there might be any updates on this yet, @RealAngeleno? Happy to make a new issue to track it if it's a bit separate from this original topic. Thanks!

**Comment by RealAngeleno** [comment]
Yeah vichan's built-in md5 hashing doesn't really work well with this. I do use a (rather jank however, but not as jank as before) method to use imagehash on python and therefore use perceptual hashes instead, which works wonders, but haven't added because it requires users to install yet another thing and because it can be slow. There is a PHP implementation I was given from an IB, though from what I've seen it's not very good at doing perceptual hashing and is also slow.
